### DISCUSSION TITLE
Custom link in site page based on circuit information

### DISCUSSION THREAD

**Discussion by tux-mania** [netbox-community/netbox#9380] [Q&A] [answered]
Hi all,

I'm unable to find a solution for this.
Basically, I have sites with circuits assigned to them.
What I would like to do is: if a specific type of circuit is assigned to a site, display a custom link **on the site page**.

I manage to create custom links for circuits displayed on circuits, and custom links for sites displayed on sites.
But I do not know how to cross information between sites and assigned circuits or devices.
Any ideas please?

Thanks

**Comment by drmsoffall** [comment, marked as answer]
It sounds like this is a true or false question: does a given site have any circuits of type X?  If so, here's a condition you can use in your link text template to show or hide your link:

`{% if obj.circuit_terminations.filter(circuit__type__name='X').count() > 0 %}My Link{% endif %}`

You can make this more efficient by finding the circuit type ID you want, and taking '__name' off the end of the query condition.  If your type ID is 1:

`{% if obj.circuit_terminations.filter(circuit__type=1).count() > 0 %}My Link{% endif %}`

**Reply by tux-mania** [reply]
Thank you very much! It works great!
But: how can I find all the properties available for an object please? I did guess some with the API, but didn't know about the 'obj.circuit_terminations' for example. 
In my case, I also need to check value of a custom field in circuit and do not know the syntax to access this object property.

Thanks for your feedback.

**Reply by drmsoffall** [reply]
For exploring the data model and testing filters, I use the [NetBox Shell](https://docs.netbox.dev/en/stable/administration/netbox-shell/).  Connect to your server, change to the NetBox directory, activate the virtual environment and then run the command at the top of the document.  It's an interactive Python shell with access to NetBox and its data, so you can use Python's `dir(x)` for example to see all the properties/methods on an object.

**Reply by tux-mania** [reply]
Thank you VERY much for your solution and this explanation! Will help me a lot in the future!
Cheers
